staging/cse/windows/provisioningscripts/securetlsbootstrap.ps1

```diff
@@ -0,0 +1,122 @@
+Param(
+    [Parameter(Mandatory=$true)][string]
+    $KubeDir,
+    [Parameter(Mandatory=$true)][string]
+    $APIServerFQDN,
+    [Parameter(Mandatory=$true)][string]
+    $AADResource,
+    [Parameter(Mandatory=$false)][string]
+    $BootstrapClientPath = [Io.path]::Combine("$KubeDir", "tls-bootstrap-client.exe"),
+    [Parameter(Mandatory=$false)][string]
+    $KubeconfigPath = [Io.path]::Combine("$KubeDir", "config"),
+    [Parameter(Mandatory=$false)][string]
+    $ClientCertPath = [Io.path]::Combine("$KubeDir", "client.crt"),
+    [Parameter(Mandatory=$false)][string]
+    $ClientKeyPath = [Io.path]::Combine("$KubeDir", "client.key"),
+    [Parameter(Mandatory=$false)][string]
+    $AzureConfigPath = [io.path]::Combine($KubeDir, "azure.json"),
+    [Parameter(Mandatory=$false)][string]
+    $ClusterCAFilePath = [io.path]::Combine($KubeDir, "ca.crt"),
+    [Parameter(Mandatory=$false)][string]
+    $LogFilePath = [Io.path]::Combine("$KubeDir", "securetlsbootstrap.log")
+)
+
+# Number of seconds to wait between bootstrap client retries
+$global:RetryDelaySeconds = 3
+
+# Value of the ALPN header to send within requests to the bootstrap server
+$global:NextProtoValue = "aks-tls-bootstrap"
+
+# Directory where we log guest-agent event telemetry for reporting/signals
+$global:EventsLoggingDir = "C:\WindowsAzure\Logs\Plugins\Microsoft.Compute.CustomScriptExtension\Events\"
+
+function CurrentUnixTimeSeconds {
+    return [DateTimeOffset]::Now.ToUnixTimeSeconds()
+}
+
+filter Timestamp { "$(Get-Date -Format o): $_" }
+
+function Write-Log($message) {
+    $msg = $message | Timestamp
+    Write-Output $msg
+}
+
+function As-Event {
```

Comment:
could you wrapper this function to a common function to call it in non-cse scripts? We have a function `Logs-To-Event` to be called in cse scripts.